src/os/src/vxworks/os_platform_errno.c

```diff
@@ -39,44 +39,49 @@ strerrorIf(
    details. NAME_MAX is defined in limits.h. */
 
 int
-os_strerror_r(int err, char *str, size_t len)
+os_strerror_r(int errnum, char *buf, size_t buflen)
 {
-    int res = EINVAL;
-
-    assert(str != NULL);
-
-    if (len < (NAME_MAX + 1)) {
-        res = ERANGE;
-    } else if (strerrorIf(err) != NULL) {
-        (void)strerror_r(err, str);
+    int err;
+    const char *str;
+
+    assert(buf != NULL);
+
+    if ((err = os_errstr(errnum, buf, buflen)) == EINVAL) {
+        err = 0;
+        if (buflen < (NAME_MAX + 1)) {
+            err = ERANGE;
+        } else if (strerrorIf(errnum) != NULL) {
+            (void)strerror_r(errnum, buf);
+        }
     }
 
-    return res;
+    return err;
 }
 
 #else
 
 int
-os_strerror_r(int err, char *str, size_t len)
+os_strerror_r(int errnum, char *buf, size_t buflen)
 {
-    int res = 0;
-
-    assert(str != NULL);
-
-    /* VxWorks's strerror_r always returns 0 (zero), so the only way to decide
-       if the error was truncated is to check if the last position in the
-       buffer is overwritten by strerror_r. */
-    str[len - 1] = 'x';
-
-    (void)strerror_r (err, str, len);
-
-    if (str[len - 1] != 'x') {
-        res = ERANGE;
+    int err;
+    const char *str;
+
+    assert(buf != NULL);
+
+    if ((err = os_errstr(errnum, buf, buflen)) == EINVAL) {
+        /* VxWorks's strerror_r always returns 0 (zero), so the only way to
```

Comment:
Alternatively, just always fill buf as well as one can, return a pointer to it and don't worry about ``EINVAL`` or ``ERANGE``. That definitely would be my preference.